scripts/babel-relay-plugin/src/RelayQLPrinter.js

```diff
@@ -534,12 +534,35 @@ module.exports = function(t: any, options: PrinterOptions): Function {
           value.map(element => this.printArgumentValue(element))
         );
       }
-      return codify({
-        kind: t.valueToNode('CallValue'),
-        callValue: t.valueToNode(value),
-      });
+      if(true||!value.hasOwnProperty("0")){
+        return codify({
+          kind: t.valueToNode('CallValue'),
+          callValue: this.printObject(value)
+        });
+      }else {
+        return codify({
+          kind: t.valueToNode('CallValue'),
+          callValue: t.valueToNode(value)
+        });
+      }
     }
 
+    // todo array!
```

Comment:
Yeah, `objectify` already implements most of `printObject`. Let's just change the core line:

    properties.push(property(key, t.valueToNode(value)));

To check the type of `value` and recurse if it is a nested object/array. Note that any helper functions can be moved out of the class definition to the bottom of the file.